### ISSUE TITLE
Error: Could not set `null` as the transaction hash (deploying with `{overwrite: false}` ) 

### ISSUE THREAD

**Issue by elie222** [trufflesuite/truffle#836]
Any ideas why this is happening in Truffle 4.1.0?

```
⚡ truffle test
Using network 'development'.

Error: Could not set `null` as the transaction hash for COT
```

COT inherits from StandardToken and doesn't do much else.

## Environment

* Operating System: OSX
* Ethereum client:
* Truffle version (`truffle version`): Truffle v4.1.0 (core: 4.1.0) Solidity v0.4.19 (solc-js)
* node version (`node --version`): v8.4.0
* npm version (`npm --version`): 5.7.1

**Comment by elie222** [comment]
This may well be a bug with Truffle. I got things working by removing: `overwrite: false` from my deploy code.

**Comment by cgewecke** [comment]
@elie222 Could you link to or show the code that results in this error, including the 'deploy code'? It would also be helpful if you provided steps to reproduce it.

**Comment by cgewecke** [comment]
@elie222 I'm going to close this because it's quite difficult from your issue description to tell what state your contracts are in when this error occurs. Feel free to re-open when you can provide more detail.

**Comment by elie222** [comment]
Thanks, I believe the issue was buggy code on our end. But as always the
error messages can be very hard to decipher.

**Comment by elie222** [comment]
Just ran into this same issue again unfortunately. Funny how when I googled it this was the top response :(

This is a sample of some of the problematic code:

```
deployer
        .deploy(Token, TokenFactory.address, { overwrite: false })
        .then(function() {
          return Token.deployed()
        })
```

**Comment by elie222** [comment]
Again the solution was to get rid of `{ overwrite: false }`, but this really wasn't ideal for me. Had to put the code into a single file to do this when I really wanted it in two separate files.

**Comment by kriskelly** [comment]
I'm also seeing this issue when using `{overwrite: false}` in my deploy code. I've verified that it deploys fine without that option.

Using Truffle v1.4.3, Solidity v0.4.19

**Comment by JackPickering** [comment]
Deploying code in separate files is not an option for me, my codebase is too big to deploy at once and therefore, I need to be able to deploy in stages. Is there a valid work around for this? 

It is the same error with me, I have {overwrite: false} in my contracts and libraries as like I said, I cant deploy all of my code at once. 

Migrations 1: 

```
var BytesDeserializer = artifacts.require("BytesDeserializer");
var StagesLib = artifacts.require("SaleStagesLib");

module.exports = async function(deployer, network, accounts) {
  deployer.deploy(StagesLib, {overwrite: false});
  deployer.deploy(BytesDeserializer, {overwrite: false});
};
```

Migrations 2:

```
var PickToken = artifacts.require("PickToken");

module.exports = async function(deployer, network, accounts) {
  deployer.deploy(PickToken, {overwrite: false});
};
```

Migrations 3:

```
var BytesDeserializer = artifacts.require("BytesDeserializer");
var StagesLib = artifacts.require("SaleStagesLib");
var KYCCrowdsale = artifacts.require("KYCCrowdsale");
var PickCrowdsale = artifacts.require("PickCrowdsale");
var PickToken = artifacts.require("PickToken");

module.exports = async function(deployer, network, accounts) {
  var sale;

  deployer.deploy(StagesLib, {overwrite: false});
  deployer.link(StagesLib, KYCCrowdsale);
  deployer.deploy(BytesDeserializer, {overwrite: false});
  deployer.link(BytesDeserializer, KYCCrowdsale);

  deployer.deploy(PickToken, {overwrite: false}).then(function() {
    return deployer.deploy(KYCCrowdsale, startTime, endTime, rate, wallet, beneficiary, buyer, founders, bounty, softCap, hardCap, PickToken.address);
  });
};
```

**Comment by cgewecke** [comment]
@JackPickering Thanks, re-opening. Will look into this and see what's going wrong . . .

**Comment by cgewecke** [comment]
@elie222 @JackPickering Apologies, it looks we introduced this bug in `4.0.7`. Have opened a PR at `truffle-contract` (ref above) which will hopefully resolve this. Should be included in a patch release scheduled for early next week.

**Comment by gnidan** [comment]
Fix for this has been released in Truffle v4.1.6. Closing for issue maintenance. Let us know if the problem persists! Thanks!

**Comment by eepstein** [comment]
That's not a patch.  That's a new release with other things changed.  Borked.
